pkg/apis/pravega/v1beta1/pravega.go

```diff
@@ -380,6 +486,53 @@ func (s *PravegaSpec) withDefaults() (changed bool) {
 		changed = true
 		s.SegmentStorePodAnnotations = map[string]string{}
 	}
+	if s.ControllerProbes == nil {
+		changed = true
+		s.ControllerProbes = &Probes{}
+	}
+	if s.ControllerProbes.ReadinessProbe == nil {
+		changed = true
+		s.ControllerProbes.ReadinessProbe = &Probe{}
+		s.ControllerProbes.ReadinessProbe.InitialDelaySeconds = DefaultControllerReadinessProbeInitialDelaySeconds
+		s.ControllerProbes.ReadinessProbe.PeriodSeconds = DefaultControllerReadinessProbePeriodSeconds
+		s.ControllerProbes.ReadinessProbe.FailureThreshold = DefaultControllerReadinessProbeFailureThreshold
+		s.ControllerProbes.ReadinessProbe.SuccessThreshold = DefaultControllerReadinessProbeSuccessThreshold
+		s.ControllerProbes.ReadinessProbe.TimeoutSeconds = DefaultControllerReadinessProbeTimeoutSeconds
+
+	}
+
+	if s.ControllerProbes.LivenessProbe == nil {
+		changed = true
+		s.ControllerProbes.LivenessProbe = &Probe{}
+		s.ControllerProbes.LivenessProbe.InitialDelaySeconds = DefaultControllerLivenessProbeInitialDelaySeconds
+		s.ControllerProbes.LivenessProbe.PeriodSeconds = DefaultControllerLivenessProbePeriodSeconds
+		s.ControllerProbes.LivenessProbe.FailureThreshold = DefaultControllerLivenessProbeFailureThreshold
+		s.ControllerProbes.LivenessProbe.SuccessThreshold = DefaultControllerLivenessProbeSuccessThreshold
+		s.ControllerProbes.LivenessProbe.TimeoutSeconds = DefaultControllerLivenessProbeTimeoutSeconds
```

Comment:
The code might be more clear if written this way (as we are overwriting all of the settings, not some of them):
```
                s.ControllerProbes.LivenessProbe = &Probe{
		    InitialDelaySeconds: DefaultControllerLivenessProbeInitialDelaySeconds,
		    PeriodSeconds: DefaultControllerLivenessProbePeriodSeconds,
		    FailureThreshold: DefaultControllerLivenessProbeFailureThreshold,
		    SuccessThreshold: DefaultControllerLivenessProbeSuccessThreshold,
		    TimeoutSeconds: DefaultControllerLivenessProbeTimeoutSeconds,
                }
```

Reply:
In helm charts we can set default value if nothing is provided

Reply:
Sounds good!